lib/vfscore/README.md

```diff
@@ -0,0 +1,148 @@
+# VFSCore: Unikraft's Virtual File System API
+
+`vfscore` (*Virtual File System Core*) is an internal library of Unikraft that provides a generic interface of system calls related to filesystem management.
+There are multiple types of filesystems available in Unikraft (e.g: `ramfs`, `9pfs`).
+
+## Organization
+
+The `vfscore` library consists of multiple files:
+
+```console
+$ tree -L 1 lib/vfscore/
+lib/vfscore/
+[...]
+├── fd.c
+├── file.c
+├── fops.c
+├── include
+├── lookup.c
+├── main.c
+├── stdio.c
+├── syscalls.c
+├── task.c
+├── vfs.h
+└── vnode.c
+```
+
+- `vnode.c` implements functions that are used to interact with the `fdtable`
+
+- `fops.c` implements the subset of functions that are defined in `vfs.h`, that are related with file operations
+
+- `lookup.c` implements the subset of functions that are defined in `vfs.h`, that are related with searching for a specific path in a filesystem
+
+- `stdio.c` exposes the functions that are used to interact with standard I / O. We can notice that this is done very similar to defining a new filesystem, as we'll discuss later in the README
+
+- `rootfs.c` implements the function that is used to mount the root filesystem
+
+- `task.c` implements some helper function that are used to convert between different path conventions
+
+- `syscalls.c` does the actual generic implementation of the `sys_*` function defined in the `vfs.h`
+
+- `main.c` defines the specific Unikraft system calls for interacting with a filesystem, which are based on the files described above. You can read more about adding a new system call for Unikraft [here](https://unikraft.org/community/hackathons/sessions/bincompat/)
```

Comment:
```suggestion
- `main.c` defines the specific Unikraft system calls for interacting with a filesystem, which are based on the files described above.
   You can read more about adding a new system call for Unikraft on the [website](https://unikraft.org/community/hackathons/sessions/bincompat/)
```

Reply:
I think [this link](https://unikraft.org/docs/develop/syscall-shim/) would be more suited here instead of the hackathon session.